### PR TITLE
SSR guide

### PR DESCRIPTION
**Pull request by NataliaTepluhina**
## Description of Problem

This is the first PR to start with Server side rendering guide. This particular one covers only the most basic SSR - without templating, without any bundler. I plan to extend this section in the follow-up PRs.

### CODE REVIEW

**Review by skirtles-code — CHANGES_REQUESTED** [review]
I've made a few small suggestions inline.

I had two other thoughts:

1. If we're going to merge this as we go then we should probably have a disclaimer at the top of each page warning that the guide is incomplete.
2. I got a little bit confused by the Express example. The example itself is fine but it doesn't include any client-side hydration. There should probably be a note somewhere that we're only dealing with the server part for now and hydration will be covered later.

On a loosely related note, the API Reference for `mount` suggests that it takes a second argument called `isHydrate`. When I tried playing with SSR myself that threw me for a while. As far as I can tell that isn't true, it only takes a single argument and you just use `createSSRApp` in the client code to trigger hydration. Perhaps something to update when the relevant SSR bits are added to the API Reference.

File: src/guide/ssr/introduction.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+If you're using [webpack](https://webpack.js.org/), you can add prerendering with the [prerender-spa-plugin](https://github.com/chrisvfritz/prerender-spa-plugin). It's been extensively tested with Vue apps.
+
+## About This Guide
+
+[//]: # 'TODO: This guide is focused on server-rendered Single-Page Applications using Node.js as the server. Mixing Vue SSR with other backend setups is a topic of its own and briefly discussed in a [dedicated section].'
+
+This guide will be very in-depth and assumes you are already familiar with Vue.js itself, and have a decent working knowledge of Node.js and webpack. If you prefer a higher-level solution that provides a smooth out-of-the-box experience, you should probably give [Nuxt.js](https://nuxtjs.org/) a try. It's built upon the same Vue stack but abstracts away a lot of the boilerplate, and provides some extra features such as static site generation. However, it may not suit your use case if you need more direct control of your app's structure. Regardless, it would still be beneficial to read through this guide to understand better how things work together.
```
[the remark this answers is not in the document]
Comment (NataliaTepluhina) [reply src/guide/ssr/introduction.md]: Commented this section

File: src/guide/ssr/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+- It's recommended to use Node.js version 10+.
+- `@vue/server-renderer` and `vue` must have matching versions.
+- `@vue/server-renderer` relies on some Node.js native modules and therefore can only be used in Node.js. We may provide a simpler build that can be run in other JavaScript runtimes in the future.
+
+## Rendering a Vue Instance
+
+Unlike Vue client-only application, SSR one should use a different method for creating an application instance: instead of `createApp` we need to use `createSSRApp`.
+
+```js
+// server.js
```
[the remark this answers is not in the document]
Comment (NataliaTepluhina) [reply src/guide/ssr/getting-started.md:26]: Removed it altogether as it's a bit early to explain about Node code before we dropped an Express name

File: src/guide/ssr/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  },
+  template: `<div>Current user is: {{ user }}</div>`
+})
+
+const appContent = await renderToString(app)
+```
+
+## Integrating with a Server
+
+To run an application, [Express](https://expressjs.com/):
```
[the remark this answers is not in the document]
Comment (NataliaTepluhina) [reply src/guide/ssr/getting-started.md]: Extended a bit

**Comment by NataliaTepluhina** [comment]
@skirtles-code I've added a few sentences and a disclaimer to show the guide is incomplete. As for hydration: it seems in the previous guide it was covered in webpack build, do you think we should have it before?

**Comment by skirtles-code** [comment]
@NataliaTepluhina I'm not sure how to cover hydration without a build. I tried to create the simplest example I could that used client-side hydration but I don't think it would make for a good documentation example:

https://github.com/skirtles-code/vue-3-ssr-example-no-build

In practice I can't imagine anyone wanting to do it that way and the tricks I used to get it to work are more likely to confuse someone rather than clarifying anything.

I think the paragraph you've added to the end of the current page explaining that client-side hydration will be explained later is sufficient to avoid confusion.

**Review by skirtles-code — APPROVED** [review]

**Pull request merged**